packages/notifications/push/amplify_push_notifications/test/amplify_push_noitfications_impl_test.dart

```diff
@@ -0,0 +1,444 @@
+        tokenReceivedEventChannel.codec.encodeErrorEnvelope(
+          code: 'test',
+          message: 'error',
+        ),
+        (_) {},
+      );
+    }
+    return null;
+  }
+
+  final config = AmplifyConfig.fromJson(
+    jsonDecode(amplifyconfig) as Map<String, Object?>,
+  );
+
+  setUp(() {
+    testWidgetsFlutterBinding = TestWidgetsFlutterBinding.ensureInitialized();
+    plugin = AmplifyPushNotifications(
+      serviceProviderClient: mockServiceProviderClient,
+      backgroundProcessor: () async => {},
+      dependencyManager: dependencyManager,
+    );
+
+    testWidgetsFlutterBinding.defaultBinaryMessenger.setMockMethodCallHandler(
+      MethodChannel(tokenReceivedEventChannel.name),
+      handler,
+    );
+  });
+
+  tearDown(log.clear);
+
+  group('Push Notifications config', () {
+    setUp(() {
+      testWidgetsFlutterBinding = TestWidgetsFlutterBinding.ensureInitialized();
+      plugin = AmplifyPushNotifications(
+        serviceProviderClient: mockServiceProviderClient,
+        backgroundProcessor: () async => {},
+        dependencyManager: dependencyManager,
+      );
+
+      testWidgetsFlutterBinding.defaultBinaryMessenger.setMockMethodCallHandler(
+        MethodChannel(tokenReceivedEventChannel.name),
+        handler,
+      );
+    });
+    test('should configure correctly', () async {
+      log.clear();
+      when(mockPushNotificationsHostApi.getLaunchNotification()).thenAnswer(
+        (_) => Future(() => standardAndroidPushMessage),
+      );
+      await plugin.configure(
+        authProviderRepo: authProviderRepo,
+        config: config,
+      );
+      expect(log, <Matcher>[
+        isMethodCall('listen', arguments: null),
+        isMethodCall('cancel', arguments: null),
+        isMethodCall('listen', arguments: null),
+      ]);
+      verify(
+        mockServiceProviderClient.recordNotificationEvent(
+          eventType: anyNamed('eventType'),
+          notification: anyNamed('notification'),
+        ),
+      ).called(1);
+    });
+
+    test('should throw exception when configuring if there is no appId present',
+        () async {
+      final config0 = AmplifyConfig.fromJson(
+        jsonDecode(noPushAppIdAmplifyConfig) as Map<String, Object?>,
+      );
+      expect(
+        () async => plugin.configure(
+          authProviderRepo: authProviderRepo,
+          config: config0,
+        ),
+        throwsA(const TypeMatcher<PushNotificationException>()),
+      );
+    });
+
+    test('should fail configure when registering device is unsuccessfull',
+        () async {
+      log.clear();
+      testWidgetsFlutterBinding.defaultBinaryMessenger.setMockMethodCallHandler(
+        MethodChannel(tokenReceivedEventChannel.name),
+        throwErrorHandler,
+      );
+      expect(
+        () async => plugin.configure(
+          authProviderRepo: authProviderRepo,
+          config: config,
+        ),
+        throwsA(const TypeMatcher<PushNotificationException>()),
```

Comment:
Should be checking more than just that it throws, e.g.

```suggestion
        throwsA(
          isA<PushNotificationException>()
            .having((e) => e.message, 'message', contains('token'),
        ),
```